scripts/dockerfiles/run.sh

```diff
@@ -85,6 +90,67 @@ done
         /var/vcap/jobs/{{ $job.Name }}/bin/run
     {{ end }}
 {{ else }}
+    # Run all the scripts called post-start, if any.
+    function post-start-paths()
+    {
+	declare -a fnames
+	idx=0
+	for fname in $(find /var/vcap/jobs/*/bin -name post-start) ; do
+	    fnames[$idx]=$fname
+	    idx=$((idx + 1))
+	done
+	echo ${fnames[*]}
+    }
+
+    psp=$(post-start-paths)
+
+    if [ "X$psp" != X ] ; then
+	# We have post-start scripts to run.
+	# We
+	# - Start a transient monit in the background, treat it as
+	#   non-background however
+	# - Wait for all jobs (except `ready`) to be ok.
+	# - Run the post-start scripts.
+	# - Get rid of the transient monit
+	# - Mark job `ready` as ok, and start the final monit,
+	#   replacing ourselves.
+	# The last two operations are also done when we have no
+	# post-start scripts.
+
+	function monit_report_down() {
+	    # 'monit report down' requires monit 5.18+. BOSH uses monit 5.6.
+	    monit status|grep status|grep -v monitoring|grep -v 'Accessible\|Running'|wc -l
+	}
+
+	dumb-init -- monit -vI &
+	sleep 10
+	while [ $(monit_report_down) -gt 1 ] ; do
+	    echo Still waiting for jobs ...
+	    sleep 10
+	done
+
+	for fname in ${psp} ; do
+	    echo bash $fname
+	    bash $fname
+	done
+
+	# Killing the monit demon now, with all jobs running does not
+	# kill the jobs, only monit itself.
+	monit quit
```

Comment:
Hm. ...  Ok. Let me see if I can explain how I came here ...

The original idea was to start monit earlier, then do the waiting, the post-start scripts, and then enter a long-running sleep wrapped into an infinite loop (like `while true;do sleep 100d;done`).

Then I  came to the comment `Replace bash with monit to handle both SIGTERM and SIGINT`
and realized that this loop I intended to use likely would not handle the signals properly.

But the existing command would.

To reconcile **start monit early**, and **keep the existing monit command for signals** I came up with **make the earlier monit transient**, i.e. start it, use it to wait for the processes to be up, then kill it after running post-start and transition over to the permanent monit under `dumb-init` to handle the signals.

Hopefully this was good enough to explain my reasoning which made the code as it currently is in the PR.